lib/aws/client.ex

```diff
@@ -5,7 +5,27 @@ defmodule AWS.Client do
 
   defstruct access_key_id: nil,
             secret_access_key: nil,
+            token: nil,
             region: nil,
-            endpoint: nil,
-            service: nil
+            endpoint: "amazonaws.com",
+            service: nil,
+            expires_at: nil
+
+  @type t :: %AWS.Client{}
+
+  @spec merge(AWS.Client.t, AWS.Client.t) :: AWS.Client.t
+  def merge(a, b) do
```

Comment:
Heh, thanks! I knew there must be some way to pattern match on anonymous functions! :+1: